### PR TITLE
feat: implement array_ndims with e2e test

### LINKED ISSUE #8881: feat: implement array_ndims
array_ndims ( anyarray ) → integer

### PR DESCRIPTION
**Pull request by Honeta**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

close #8881

## Checklist For Contributors

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Checklist For Reviewers

- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.

## Documentation

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- SQL commands, functions, and operators

### Release note

</details>

### CODE REVIEW

**Review by st1page — APPROVED** [review]
LGTM

File: src/expr/src/expr/expr_array_ndims.rs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+/// select array_ndims(array[NULL]);
+/// ----
+/// 1
+///
+/// query error unknown type
+/// select array_ndims(null);
+/// ```
+#[function("array_ndims(list) -> int64")]
+fn array_ndims(array: ListRef<'_>) -> i64 {
+    if let Some(ScalarRefImpl::List(list_ref)) = array.value_at(1).unwrap() {
```
[the remark this answers is not in the document]
Comment (xxchan) [reply src/expr/src/expr/expr_array_ndims.rs:63]: image
It's safe 😅

[the remark this answers is not in the document]
Comment (xxchan) [reply src/expr/src/expr/expr_array_ndims.rs:63]: this implementation looks correct, although might be different from PG

File: src/expr/src/expr/expr_array_ndims.rs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+/// ```
+///
+/// Examples:
+///
+/// ```slt
+/// query T
+/// select array_ndims(null::int[]);
+/// ----
+/// NULL
+///
```
Comment (xxchan) [inline src/expr/src/expr/expr_array_ndims.rs:32]:
```suggestion
///
/// ```slt
/// select array_ndims(array[]::int[]);
/// ----
/// NULL
```
One edge case from PG. Currently this returns 1 in the implementation. (Although I'm not very uncomfortable about the difference..

File: src/expr/src/expr/expr_array_ndims.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (st1page) [reply src/expr/src/expr/expr_array_ndims.rs:63]: I prefer to wait for deciding our array's behavior @xiangjinwu and see if we can implement it in optimizer 🤔

**Pull request closed without merging**